errata.go

```diff
@@ -0,0 +1,51 @@
+package multihash
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"hash"
+)
+
+type identityMultihash struct {
+	bytes.Buffer
+}
+
+func (identityMultihash) BlockSize() int {
+	return 32 // A prefered block size is nonsense for the "identity" "hash".  An arbitrary but unsurprising and positive nonzero number has been chosen to minimize the odds of fascinating bugs.
+}
+
+func (x identityMultihash) Size() int {
+	return x.Len()
+}
+
+func (x identityMultihash) Sum(digest []byte) []byte {
+	return x.Bytes()
+}
+
+type doubleSha256 struct {
+	main hash.Hash
+}
+
+func (x doubleSha256) Write(body []byte) (int, error) {
+	return x.main.Write(body)
+}
+
+func (doubleSha256) BlockSize() int {
+	return sha256.BlockSize
+}
+
+func (doubleSha256) Size() int {
+	return sha256.Size
+}
+
+func (x doubleSha256) Reset() {
+	x.main.Reset()
+}
+
+func (x doubleSha256) Sum(digest []byte) []byte {
+	intermediate := [sha256.Size]byte{}
+	x.main.Sum(intermediate[0:0])
+	h2 := sha256.New()
+	h2.Write(intermediate[:])
```

Comment:
1. If we're going to allocate, why not just `sha256.Sum256(intermediate)`?
2. Otherwise, we can probably just re-use the digest instead of creating a new intermediate.

Reply:
Hm.  Looked at (1), but I think it's slightly, minutely worse to do that because then we can't hand down our `digest` byte slice parameter we were given for output space (and using that slice instead of introducing `intermediate` would be cost-neutral, but isn't a good option either because it would be a break of what people expect Sum to do -- if Sum mutates its parameter it's supposed to be to set it to the result, and I'd be nervous about doing anything else).

I thought about (2), too, but it would make our Sum method side-effecting (ex: two consecutive calls would yield different results) which would also be a break of what Sum is expected to do.

Reply:
> I thought about (2), too, but it would make our Sum method side-effecting (ex: two consecutive calls would yield different results) which would also be a break of what Sum is expected to do.

No?

Reply:
... Right, I figured out the correct sprinkling of `[0:0]`s.  I was clearly too tired when I looked at this last time.  It now uses the `digest` memory repeatedly.

This wasn't my original goal or focus, but I believe this is now significantly _improved_ from the previous implementation, which returned new slices in at least two places.